tests/framework/cli/micropkg/test_micropkg_pull.py

```diff
@@ -17,6 +17,14 @@
 PIPELINE_NAME = "my_pipeline"
 
 
+@pytest.fixture
+def temp_dir_with_context_manager(tmp_path):
+    mock_temp_dir = Mock()
+    mock_temp_dir.__enter__ = Mock(return_value=tmp_path)
+    mock_temp_dir.__exit__ = Mock(return_value=None)
+    return mock_temp_dir
+
+
```

Comment:
The issue is in micropkg  `pull_package` we are using ContextManager i.e., using `with tempfile.TemporaryDirectory()`  . While mocking in the tests we are creating a mock directory which does not support ContextManager. To support it, we need to have __enter__ and __exit__ methods defined. So in the updated tests, I created a fixture to reuse. Thank you

Reply:
Why is that only happening in 3.13?

Reply:
As discussed in chat, the issue was also happening in python 3.9 [build](https://github.com/kedro-org/kedro/actions/runs/11819838376/job/32946668583) . This might be due to some library version change, which I could not find.

Reply:
I think that's fine as this only break tests, and very few are using the feature (it will be deprecated in 0.20).